btk/metrics.py

```diff
@@ -1,484 +1,110 @@
-    # find distance to nearest neighbor. If isolated then set to np.inf
-    true_table["min_dist"] = np.ones(num_true) * np.inf
-    get_closest_neighbor_distance(true_table)
-    # initialize detected objects table columns
-    num_det = len(detected_table)
-    detected_table["detection_id"] = range(num_det)  # id in blend [0-num_det]
-    # index of blend in test size [0 - len(blend_summary)]
-    detected_table["blend_index"] = np.ones(num_det, dtype=int) * (
-        batch_index * batch_size + blend_index
-    )
-    # index of batch in test size [0 - batch_size]
-    detected_table["batch_index"] = np.ones(num_det, dtype=int) * batch_index
-    # id of closest true object; [0 - num_true] if detected, else -1
-    detected_table["match_true_id1"] = np.ones(num_det, dtype=int) * -1
-    detected_table["match_true_id2"] = np.ones(num_det, dtype=int) * -1
-    detected_table["match_galtileid1"] = np.ones(num_det, dtype=int) * -1
-    detected_table["match_galtileid2"] = np.ones(num_det, dtype=int) * -1
+import skimage.metrics
 
 
 def get_detection_match(true_table, detected_table):
     """Match detections to true objects and update values in the input
     blend catalog and detection catalog.
-
     Function does not return anything, only the astropy tables are updated.
-
     Args:
         true_table (astropy.table.Table): Table with entries corresponding to
             the true object parameter values in one blend.
         detected_table(astropy.table.Table): Table with entries corresponding
             to output of measurement algorithm in one blend.
     """
+    match_table = astropy.table.Table()
     if len(detected_table) == 0 or len(true_table) == 0:
         # No match since either no detection or no true objects
         return
     t_x = true_table["x_peak"][:, np.newaxis] - detected_table["x_peak"]
     t_y = true_table["y_peak"][:, np.newaxis] - detected_table["y_peak"]
     dist = np.hypot(t_x, t_y)
-    norm_size = true_table["size"]
-    norm_dist = dist / norm_size[:, np.newaxis]
-    detected_table["dSigma_min"] = np.min(norm_dist, axis=0)
     detected_table["d_min"] = np.min(dist, axis=0)
-    detection_threshold1 = 5
-    condlist1 = [
-        np.min(dist, axis=0) <= detection_threshold1,
-        np.min(dist, axis=0) > detection_threshold1,
+    detection_threshold = 5
+    condlist = [
+        np.min(dist, axis=0) <= detection_threshold,
+        np.min(dist, axis=0) > detection_threshold,
     ]
-    choicelist1 = [np.argmin(dist, axis=0), -1]
-    match_id1 = np.select(condlist1, choicelist1)
-    detected_table["match_true_id1"] = match_id1
-    detected_table["match_galtileid1"] = true_table["galtileid"][match_id1]
-    detection_threshold2 = 0.5
-    condlist2 = [
-        np.min(norm_dist, axis=0) <= detection_threshold2,
-        np.min(norm_dist, axis=0) > detection_threshold2,
+    choicelist = [np.argmin(dist, axis=0), -1]
+    match_id = np.select(condlist, choicelist)
```

Comment:
I actually never used this function before, but I think it works great in this case